Makefile

```diff
@@ -1,60 +1,91 @@
 # Set default shell to bash
-SHELL := /bin/bash
+SHELL := /bin/bash -o pipefail
 
-MOUNT_TARGET_DIRECTORY  = /app/src
-BUILD_TOOLS_DOCKER_REPO = mineiros/build-tools
+BUILD_TOOLS_VERSION      ?= v0.5.4
+BUILD_TOOLS_DOCKER_REPO  ?= mineiros/build-tools
+BUILD_TOOLS_DOCKER_IMAGE ?= ${BUILD_TOOLS_DOCKER_REPO}:${BUILD_TOOLS_VERSION}
 
-# Set default value for environment variable if there aren't set already
-ifndef BUILD_TOOLS_VERSION
-	BUILD_TOOLS_VERSION := v0.5.3
-endif
+# if running in CI (e.g. Semaphore CI)
+# https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#ci
+#
+# to disable TF_IN_AUTOMATION in CI set it to empty
+# https://www.terraform.io/docs/commands/environment-variables.html#tf_in_automation
+#
+# we are using GNU style quiet commands to disable set V to non-empty e.g. V=1
+# https://www.gnu.org/software/automake/manual/html_node/Debugging-Make-Rules.html
+#
+ifdef CI
+	TF_IN_AUTOMATION ?= yes
+	export TF_IN_AUTOMATION
 
-ifndef BUILD_TOOLS_DOCKER_IMAGE
-	BUILD_TOOLS_DOCKER_IMAGE := ${BUILD_TOOLS_DOCKER_REPO}:${BUILD_TOOLS_VERSION}
+	V ?= 1
 endif
 
-ifndef DOCKER_SOCKET
-	DOCKER_SOCKET := /var/run/docker.sock
+ifndef NOCOLOR
+	GREEN  := $(shell tput -Txterm setaf 2)
+	YELLOW := $(shell tput -Txterm setaf 3)
+	WHITE  := $(shell tput -Txterm setaf 7)
+	RESET  := $(shell tput -Txterm sgr0)
 endif
 
-GREEN  := $(shell tput -Txterm setaf 2)
-YELLOW := $(shell tput -Txterm setaf 3)
-WHITE  := $(shell tput -Txterm setaf 7)
-RESET  := $(shell tput -Txterm sgr0)
+DOCKER_RUN_FLAGS += --rm
+DOCKER_RUN_FLAGS += -v ${PWD}:/app/src
+DOCKER_RUN_FLAGS += -e TF_IN_AUTOMATION
+#DOCKER_RUN_FLAGS += -e USER_UID=$(shell id -u)
```

Comment:
shouldn't this be uncommented ?

Reply:
see disabled test that leaks credentials... docker in docker does not work in user mode... tests are kind of broken so reduced the scope for now..